ui/src/components/build/CreateBuildConfigDialog.js

```diff
@@ -0,0 +1,189 @@
+  KEY_CR_BUILD_IOS_CREDENTIALS,
+  KEY_CR_BUILD_IOS_CREDENTIALS_DEVELOPER_PROFILE,
+  KEY_CR_BUILD_IOS_CREDENTIALS_NAME,
+  CLIENT_TYPE_ANDROID,
+  CLIENT_TYPE_IOS,
+  KEY_HIDE_PLATDORM,
+  BUILD_PLATFORM_IOS
+} from './Constants';
+import { SubState } from '../common/SubState';
+import { renderNameSection } from './create_build_config/NameSection';
+import {
+  configValidation,
+  buildValidation,
+  sourceValidation,
+  basicAuthValidation,
+  sshAuthValidation,
+  androidCredentialsValidation,
+  iOSCredentialsValidation
+} from './create_build_config/Validations';
+import { renderSourceSection } from './create_build_config/SourceSection';
+import { renderBuildSection } from './create_build_config/BuildSection';
+import { isEqual } from 'lodash-es';
+
+class CreateBuildConfigDialog extends Component {
+  static propTypes = {
+    onCancel: PropTypes.func,
+    onSave: PropTypes.func,
+    title: PropTypes.string.isRequired,
+    show: PropTypes.bool.isRequired,
+    update: PropTypes.bool,
+    initialConfig: PropTypes.object
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      valid: false,
+      [KEY_HIDE_PLATDORM]: false,
+      config: {
+        [KEY_CR_SOURCE]: {
+          [KEY_CR_SOURCE_AUTH_TYPE]: BUILD_AUTH_TYPE_PUBLIC,
+          [KEY_CR_SOURCE_GITREF]: 'master',
+          [KEY_CR_SOURCE_JENKINS_FILE_PATH]: './'
+        },
+        [KEY_CR_BUILD]: { [KEY_CR_BUILD_PLATFORM]: BUILD_PLATFORM_ANDROID, [KEY_CR_BUILD_TYPE]: BUILD_TYPE_DEBUG },
+        ...this.props.config
+      }
+    };
+
+    this.configState = new SubState(this, 'config', configValidation, [KEY_CR_NAME]);
+    this.buildState = new SubState(this, 'config.build', buildValidation);
+    this.basicAuthState = new SubState(this, 'config.source.basicAuth', basicAuthValidation, [
+      KEY_CR_BASIC_AUTH_NAME,
+      KEY_CR_BASIC_AUTH_PASSWORD,
+      KEY_CR_BASIC_AUTH_USERNAME
+    ]);
+    this.sshAuthState = new SubState(this, 'config.source.sshAuth', sshAuthValidation, [
+      KEY_CR_SSH_AUTH_NAME,
+      KEY_CR_SSH_PRIVATE_KEY
+    ]);
+    this.sourceState = new SubState(this, 'config.source', sourceValidation, [KEY_CR_SOURCE_GITURL]);
+    this.androidCredentialsState = new SubState(this, 'config.build.androidCredentials', androidCredentialsValidation, [
+      KEY_CR_BUILD_ANDROID_CREDENTIALS_NAME,
+      KEY_CR_BUILD_ANDROID_CREDENTIALS_KEYSTORE,
+      KEY_CR_BUILD_ANDROID_CREDENTIALS_KEYSTORE_ALIAS,
+      KEY_CR_BUILD_ANDROID_CREDENTIALS_KEYSTORE_PASSWORD
+    ]);
+    this.iOSCredentialsState = new SubState(this, 'config.build.iosCredentials', iOSCredentialsValidation, [
+      KEY_CR_BUILD_IOS_CREDENTIALS,
+      KEY_CR_BUILD_IOS_CREDENTIALS_DEVELOPER_PROFILE,
+      KEY_CR_BUILD_IOS_CREDENTIALS_NAME
+    ]);
+  }
+
+  componentDidUpdate(prevProps) {
+    this.validate();
+    const { initialConfig: prevConfig = {} } = prevProps;
+    const { initialConfig = {} } = this.props;
+    if (!isEqual(prevConfig, initialConfig)) {
```

Comment:
Why do we need to to update state based on props? IMO `CreateBuildConfigDialog` should only be initializes with config (in case of buildConfig update) in constructor of this component.

Reply:
removed